### ISSUE TITLE
[question] How do I properly wipe a disconnected session?

### ISSUE THREAD

**Issue by diwic** [moscajs/aedes#473]
Hi!

Given that MQTTv5 session expiry is not implemented yet, what is the current best way to wipe a persistent (non-clean) session that's been disconnected for a while? 

E g, if a persistent session has been disconnected for a specific time interval, or the number of queued messages has exceeded a certain number, or something else, I want aedes to completely forget about the client/session, both its subscriptions and its current queue of offline messages.

I'm not sure if this is done in aedes or directly in the persistence layer, but if the latter, let's assume that I'm using `aedes-persistence`, i e, memory only persisence.

Thanks in advance!

**Comment by robertsLando** [comment]
@diwic What about [client.close](https://github.com/moscajs/aedes/blob/master/lib/client.js#L237)

**Comment by diwic** [comment]
@robertsLando Thanks for the quick reply!

I'm looking at the code you pointed at, and I'm not familiar with it so correct me if I'm wrong, but from what I read, it seems to disconnect the socket and unsubscribe the topics (regardless of whether the session is persistent or not), but it's not actually emptying the current queue of offline messages?

**Comment by robertsLando** [comment]
> it's not actually emptying the current queue of offline messages

Could you explain better what you mean? Do you mean outgoing messages queue?

**Comment by diwic** [comment]
While the client is disconnected, the server must accumulate/store messages (with QoS >= 1) for the client that is disconnected. I think this is the same as the "outgoing messages queue".

If there was something similar to the `emptyQueue` function you pointed to, but would simply discard all messages in the queue instead of writing them to a just connected client, I could use that together with `client.close`. Does that make sense?

Basically, I want to discard everything that's [stored in a persistent session](https://www.hivemq.com/blog/mqtt-essentials-part-7-persistent-session-queuing-messages/).

**Comment by robertsLando** [comment]
Actually there is not such method, but it could be implemented on client

**Comment by robertsLando** [comment]
@diwic Check #474

**Comment by robertsLando** [comment]
@diwic Will submit a patch release so you can test this

**Comment by diwic** [comment]
@robertsLando Thanks! No hurry though, won't have time to test until next week.
